ambari-server/src/main/java/org/apache/ambari/server/controller/logging/LoggingRequestHelperImpl.java

```diff
@@ -295,30 +296,52 @@ private String getLogSearchAdminPassword() {
     return null;
   }
 
-  public Set<String> sendGetLogFileNamesRequest(String componentName, String hostName) {
-    Map<String, String> queryParameters =
-      new HashMap<>();
+  public HostLogFilesResponse sendGetLogFileNamesRequest(String hostName) {
+    try {
+      // use the Apache builder to create the correct URI
+      URIBuilder uriBuilder = createBasicURI(protocol);
+      uriBuilder.setPath(LOGSEARCH_GET_LOG_FILES_PATH);
+      uriBuilder.addParameter(HOST_QUERY_PARAMETER_NAME, hostName);
+      uriBuilder.addParameter(LOGSEARCH_CLUSTERS_QUERY_PARAMETER_NAME, cluster.getClusterName());
+      // ask for page size of 1, since we really only want a single entry to
+      // get the file path name
+      uriBuilder.addParameter(PAGE_SIZE_QUERY_PARAMETER_NAME, "1");
```

Comment:
I think the page size query parameter can be removed now.  The original workaround to get the file names involved making a search query of page size = 1 in order to get the component/file name metadata.  

Since this new API call handles that, I'm not sure if the page size is required anymore, but I'd be inclined to think that it is not required.  

I would recommend removing the page size parameter, and retesting this.